tests/dragonfly/instance.py

```diff
@@ -74,6 +74,12 @@ def __init__(self, params: DflyParams, args):
             del self.args["logtostderr"]
             self.args["alsologtostderr"] = None
 
+        # Run with num_shards = (proactor_threads - 1) if possible, so help expose bugs
+        if "proactor_threads" in self.args and "num_shards" not in self.args:
```

Comment:
Now we only set `num_shards` when `proactor_threads` is set, but it's probably unset in most cases. The issue is figuring out the unset value.
Shall I apply this logic even if `proactor_threads` is not set? If so, to which value?

Reply:
if not set, please set it to `psutil.cpu_count()` and num_shards to -1 if the former is `>2` @chakaz

Reply:
done